### PR TITLE
Pulumi: add backup

### LINKED ISSUE #1497: Pulumi: SRE backup

### PR DESCRIPTION
**Pull request by jemrobinson**
## :white_check_mark: Checklist

- [x] You have given your pull request a meaningful title (_e.g._ `Enable foobar integration` rather than `515 foobar`).
- [x] You are targeting the appropriate branch. If you're not certain which one this is, it should be **`develop`**.
- [x] Your branch is up-to-date with the **target branch** (it probably was when you started, but it may have changed since then).
- [x] You have marked this pull request as a **draft** and added `'[WIP]'` to the title if needed (if you're not yet ready to merge).
- [x] You have formatted your code using appropriate automated tools (for example `./tests/AutoFormat_Powershell.ps1 -TargetPath <path to file or directory>` for Powershell).

### :arrow_heading_up: Summary

- Adds blob backup for secure data
- Allows for disk backup but this is not currently needed

### :closed_umbrella: Related issues

Closes #1497 

### :microscope: Tests

Tested locally

### CODE REVIEW

File: data_safe_haven/pulumi/components/sre_backup.py [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+    def __init__(
+        self,
+        name: str,
+        stack_name: str,
+        props: SREBackupProps,
+        opts: ResourceOptions | None = None,
+    ) -> None:
+        super().__init__("dsh:sre:BackupComponent", name, {}, opts)
+        child_opts = ResourceOptions.merge(opts, ResourceOptions(parent=self))
+        start_date = str(datetime.datetime.now(datetime.timezone.utc).date())
```
Comment (JimMadge) [inline data_safe_haven/pulumi/components/sre_backup.py]: `start_date` will change every time the code is run. Won't that cause Pulumi to make unnecessary changes and make it appear the state is incorrect?

What happens if the time ISO repeating time period just starts are some date in the past?

**Review by JimMadge — CHANGES_REQUESTED** [review]
Looks good.

A question/suggestion about the start date for the disk backup interval.

File: data_safe_haven/pulumi/components/sre_backup.py (hunk 1 above)
Comment (jemrobinson) [reply data_safe_haven/pulumi/components/sre_backup.py]: Tested and this works. Done

**Review by JimMadge — APPROVED** [review]
